plugins/block-dynamic-connection/src/dynamic_if.ts

```diff
@@ -198,14 +206,16 @@ Blockly.Blocks['dynamic_if'] = {
    * Called by a monkey-patched version of InsertionMarkerManager when a block
    * drag ends if the dragged block had a pending connection with this block.
    */
-  finalizeConnections: function(this: DynamicIfBlock): void {
+  finalizeConnections(this: DynamicIfBlock): void {
     const toRemove = [];
     // Remove Else If inputs if neither the if nor the do has a connected block.
     for (let i = 2; i < this.inputList.length - 1; i += 2) {
       const ifConnection = this.inputList[i];
       const doConnection = this.inputList[i + 1];
-      if (!ifConnection.connection.targetConnection &&
-          !doConnection.connection.targetConnection) {
+      if ((!ifConnection.connection ||
+              !ifConnection.connection.targetConnection) &&
+          (!doConnection.connection ||
+              !doConnection.connection.targetConnection)) {
```

Comment:
Hmm.  Why not use `?.`—e.g. `if(!ifConnection.connection?.targetConnection && …`?

I think that should be equivalent (but maybe double-check that I am not confused!)

(Here and several places below where you have added `||` clauses.)

Reply:
Done.